LEAF_Request_Portal/js/formQuery.js

```diff
@@ -223,17 +222,83 @@ var LeafFormQuery = function() {
     	successCallback = funct;
     }
 
+    /**
+     * onProgress assigns a callback to be called on every getBulkData() iteration
+     * @param funct - funct(int Progress). Progress is the number of records that have been processed
+     * @memberOf LeafFormQuery
+     */
+    function onProgress(funct) {
+        progressCallback = funct;
+    }
+
+    /**
+     * Execute search query in chunks
+     * @param limitOffset Used in subsequent recursive calls to track current offset
+     * @returns Promise resolving to query response
+     * @memberOf LeafFormQuery
+     */
+    let results = {};
+    let batchSize = 500;
+    function getBulkData(limitOffset) {
+        if(limitOffset == undefined) {
+            limitOffset = 0;
+        }
+        if(limitOffset == 0) {
+            results = {};
+        }
+
+        query.limit = batchSize;
+        query.limitOffset = limitOffset;
+
+        let el = document.createElement('div');
+        el.innerHTML = JSON.stringify(query);
+        let queryUrl = el.innerText;
+
+        let dataType = 'json';
+        let usingJSONP = '';
+        if(useJSONP) {
+            dataType = 'jsonp';
+            usingJSONP = '&format=jsonp';
+        }
+
+        return $.ajax({
+            type: 'GET',
+            url: rootURL + 'api/form/query?q=' + queryUrl + extraParams + usingJSONP,
+            dataType: dataType
+        }).then(function(res, resStatus, resJqXHR) {
+            results = Object.assign(results, res);
+
+            if(Object.keys(res).length == batchSize
+                || resJqXHR.getResponseHeader('leaf-query') == 'continue') {
+                let newOffset = limitOffset + batchSize;
```

Comment:
limitOffset could technically be equal to "plaid" here.  Recommend parseInt(limitOffset) to ensure int gets added to int.

Reply:
Added parseInt higher up